lib/shopify_transporter/pipeline/magento/order/top_level_attributes.rb

```diff
@@ -25,13 +34,111 @@ def convert(hash, record)
 
           private
 
+          ORDER_ORIGINATED_FROM = 'Magento'
+
           def build_customer(hash)
             {
               email: hash['customer_email'],
               first_name: hash['customer_firstname'],
               last_name: hash['customer_lastname'],
             }.stringify_keys
           end
+
+          def total_price(hash)
+            hash['grand_total'].to_i
+          end
+
+          def total_paid(hash)
+            return hash['total_paid'].to_i if hash['total_paid'].present?
+            0
+          end
+
+          def total_refunded(hash)
+            return hash['total_refunded'].to_i if hash['total_refunded'].present?
+            0
+          end
+
+          def paid?(hash)
+            total_price(hash) == total_paid(hash) && total_refunded(hash) == 0
+          end
+
+          def partially_paid?(hash)
```

Comment:
minor/this is a preference:
I suggest moving the method definitions for these methods to below the method where they are used. This makes it so that developers reading the code can ignore the lower level implementations unless they are interested in it.

Reply:
Good point! I've relocated all helper functions to the bottom of the class.